### ISSUE TITLE
[Question] Adding concrete types when calling a function with generics.

### ISSUE THREAD

**Issue by nadeesha** [facebook/flow#4165]
I'm coming from a typescript background and a bit confused about the use of generics in flow.

In typescript, we can do something like:

```js
function<T, V> convertToFoo(params: T): V {
   return transform(params);
}

const result = convertToFoo<ParamType, ResultType>(myParams); // result will be typeof ResultType now
```

But does flow allow you to honker down generics when you're calling a function like above?

A more real-world example is here: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjOA7AzgFzAEUBXAUwCcBPAJTOwBNLKwBeMACgEo2A+MADwMAlgDdgvdPioAHMmABiJbBnzCcAYTgBbGTnr4AKrLICAgv3YcZFODNwAuMGZ6t+AfjoBDVQBIAojBk2gYCXthUvADcqNJyYBowXri4Wrr62EYmAgAiADTOBQDKlglJKQLefml62KH5hWAl0ahkAB56FIRx8jUZWXLmpUoqapo6tQbGgxZgAD5lyakT-dOm4VQFZgUbLT2K2ACMa+YFAEKlHF5OLnxgZzH7ABJwGCfb98NHJ311me+8Aq-KbZC4tWImMB0XAkGD4AAKtnsJ2hsPwTgA3gBfAoAeQQ2ERdkcYGxBQAal4KMIvAAjIIk7GlfGEpG4MAAMlJqDAYEotgoTn8FAFeR5YAoZBhcKcqLhYt58C8ImwAHMnLS4HAguEFWBRFSafSpU5KdS6QyxViYqgoMpVOpsGAEMJ8AALUiUKgCM1GhmYnFgFlE+wBgpy9GkrG8DjigCO5GoTgI1LVeoN5uNJN9FqlYq4Theb2yEZDuBRUrReIJZYphtzuEBQZrbP4GPFkvwJAoTo4Z2SZGBmScQ4GplLbIr0vw1dZxLrmYZvF4BYSKz+Y4Ewdbd1jvN5iDnoebR9wetc-D3+4EnuodEYzAo4v3fOwomEtmwIUyrAx2Kxz77gmXq-sB1AAS+vIZn6Uq-tBDYQZBkoPhQv4cO2kH7vycAUHqmE2POgG8limJEVhIo4UKFG4WRvIEceEZ4fuWIXpwtGCP2uCDuuBikgAdAJh5llifLUb+2EUCJkrTr+9G4CJSoqqqv4AIQSZyXIqXJInwVmcH1lmIkSOxXCIbyxm8lwMQARC8SGMqXhrGw3IAJD4A5TgAHIkNotKUExUBau4yb4KmqrWegLrure3r2QwjkmAUZJRjGADkUBeAA1mQAC0YFUKlSXufFTiHCxURAA

**Comment by awestroke** [comment]
You would do it like this:

```
const result: ResultType = convertToFoo(myParams: ParamType);
```

**Comment by jakewtaylor** [comment]
+1

This could be used to improve some types included by Flow, see [#4559](https://github.com/facebook/flow/issues/4559#issuecomment-339642966)
